### ISSUE TITLE
Incorrect end offset with `import.meta.*` and dynamic imports

### ISSUE THREAD

**Issue by marvinhagemeister** [guybedford/es-module-lexer#79]
When a dynamic import expression uses `import.meta.*` the parser gets confused and returns a wrong end offset. This is reproducible with the following snippet.

```js
const [imports] = await parse(`import(import.meta.url)`);

console.log(imports);
// Logs:
// [
//   { n: undefined, s: 7, e: -7224, ss: 0, se: 0, d: 0, a: -1 },
//   { n: undefined, s: 7, e: 18, ss: 7, se: 18, d: -2, a: -1 }
// ]
```

The end position of the actual `import()` call is very wrong here. It should be positive and not `-7224`.

**Comment by guybedford** [comment]
The code in question is https://github.com/guybedford/es-module-lexer/blob/main/src/lexer.c#L117 which checks for the closing brace of the dynamic import.

The quick fix would be to track the index of the dynamic import to distinguish it from eg import.meta so that instead of checking the head we are checking the exact last dynamic import.

More generally we also don't support nested dynamic imports - `import(await import('path'))`, but perhaps we can push that one out for now as well and focus on the direct `import.meta` fix of storing the last dynamic import pointer only instead of needing a stack just yet.

Let me know if you're able to look into a PR, otherwise I will add this to "the list".

**Comment by guybedford** [comment]
Sorry I pointed to the wrong line there, should have pointed to https://github.com/guybedford/es-module-lexer/blob/main/src/lexer.c#L107.

**Comment by marvinhagemeister** [comment]
Agree, I don't thing supporting nested dynamic imports is worth it right now. I've never seen that used anyhwere so far in my bubble and I suspect that it's extremely rare. Mixing dynamic import statements seems to be getting some traction though in my circle with now Node supporting native ESM.

Doubt I can contribute much to the C portion here as I've never coded in C. I managed to get all the wasm toolkit installed in the right directories to be able to build the wasm binary and run the tests though.

The workaround we're currently using in WMR can be found here: https://github.com/preactjs/wmr/blob/7ad9461b32269a70f49786a4193342ff7e0b23c8/packages/wmr/src/lib/transform-imports.js#L38-L49

```js
const [imports] = await parse(code);

for (const item of imports) {

  // Dynamic import
  if (item.d > -1) {
    // Bugfix: `import(import.meta.url)` returns an invalid negative end_offset.
    // We detect that here and find the closing paren to estimate the offset.
    if (item.e < 0) {
      item.e = code.indexOf(')', item.s);
	  item.d = -2 // Mark as import.meta
    }

    // dynamic import() has no statement_end, so we take the position following the closing paren:
    item.se = code.indexOf(')', item.e) + 1;
  }
  
  // ...
}
```

**Comment by guybedford** [comment]
Fix released in 0.7.1.
